Test1/Test1/ThreadSafePriorityQueue.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace Test1;
+
+public class ThreadSafePriorityQueue<P, V > where P : IComparable<P>
+{
+    private readonly SortedDictionary<P, List<V>> queue = new();
+    private int count = 0;
+    private readonly object queueLock = new();
+
+    public void Enqueue(P priority, V value)
+    {
+        lock (queueLock)
+        {
+            if (queue.ContainsKey(priority) != true)
+            {
+                var list = new List<V>();
+                queue.Add(priority, list);
+            }
+            queue[priority].Add(value);
+            count++;
+            Monitor.PulseAll(queueLock);
+        }
+    }
+
+    public V Dequeue()
+    {
+        lock (queueLock)
+        {
+            if (queue.Count == 0) Monitor.Wait(queueLock);
```

Comment:
```suggestion
            while (queue.Count == 0) Monitor.Wait(queueLock);
```
А если несколько потоков заблокируются на `Wait`? Затем в очередь добавят элемент, все эти потоки проснутся. Один из них возьмет элемент из очереди, а остальные упадут на `queue.Keys.Last()`. 
Я ведь говорил на паре, что не `if`, а `while` тут нужно использовать.

Reply:
-1 балл.